Source/WebCore/editing/CompositeEditCommand.cpp

```diff
@@ -613,7 +614,10 @@ void CompositeEditCommand::insertNodeAt(Ref<Node>&& insertChild, const Position&
 
 void CompositeEditCommand::appendNode(Ref<Node>&& node, Ref<ContainerNode>&& parent)
 {
-    ASSERT(canHaveChildrenForEditing(parent));
+    // When cloneParagraphUnderNewElement() clones the fallback content of an OBJECT element,
+    // the ASSERT below may fire since the return value of canHaveChildrenForEditing is not reliable
+    // until the render object of the OBJECT is created. Hence we ignore this check for OBJECTs.
+    ASSERT(canHaveChildrenForEditing(parent) || (parent->isElementNode() && downcast<ContainerNode>(*parent).tagQName() == objectTag));
```

Comment:
A simpler check is canHaveChildrenForEditing(parent) || parent->hasTagName(objectTag).

Reply:
Sweet! Done!